apps/to-do-api/src/use-cases/get-task.ts

```diff
@@ -0,0 +1,16 @@
+import * as RTE from "fp-ts/lib/ReaderTaskEither.js";
+import { pipe } from "fp-ts/lib/function.js";
+
+import { Task } from "../domain/Task.js";
+import { getTask } from "../domain/TaskRepository.js";
+import { ItemNotFound } from "../domain/errors.js";
+
+export const getTaskById = (id: Task["id"]) =>
+  pipe(
+    getTask(id),
+    // Handle the Option
+    RTE.flatMapOption(
+      (some) => some,
+      () => new ItemNotFound("Task not found"),
+    ),
```

Comment:
nit: maybe you can use `fromOption`

```suggestion
RTE.fromOption(
  () => new ItemNotFound("Task not found")
)```

Reply:
it works!